tests/staticfiles_tests/test_checks.py

```diff
@@ -132,3 +136,57 @@ def test_nonexistent_directories(self):
             # Nonexistent directories are skipped.
             finder = get_finder("django.contrib.staticfiles.finders.FileSystemFinder")
             self.assertEqual(list(finder.list(None)), [])
+
+
+class StoragesCheckTests(SimpleTestCase):
+    @override_settings(STORAGES={})
+    def test_error_empty_storages(self):
+        # "default" and "staticfiles" need to be popped from STORAGES since
+        # UserSettingsHolder has code to maintain backward compatibility.
+        # RemovedInDjango51Warning
+        settings.STORAGES.clear()
+        assert settings.STORAGES == {}
+        errors = check_storages(None)
+        self.assertEqual(
+            errors, [Error(MISSING_STORAGE_SETTING, id="staticfiles.E005")]
+        )
+
+    def test_no_errors_storages_default_value(self):
+        assert sorted(settings.STORAGES.keys()) == ["default", "staticfiles"]
+        errors = check_storages(None)
+        self.assertEqual(errors, [])
+
+    @override_settings(
+        STORAGES={
+            "staticfiles": {
+                "BACKEND": "django.contrib.staticfiles.storage.StaticFilesStorage",
+            },
+        }
+    )
+    def test_no_errors_staticfiles_only(self):
+        # See previous comment about UserSettingsHolder compat code
+        # RemovedInDjango51Warning
+        settings.STORAGES.pop("default")
+        assert "default" not in settings.STORAGES
+        errors = check_storages(None)
+        self.assertEqual(errors, [])
+
+    @override_settings(
+        STORAGES={
+            "default": {
+                "BACKEND": "django.core.files.storage.FileSystemStorage",
+            },
+            "example": {
+                "BACKEND": "ignore.me",
+            },
+        }
+    )
+    def test_error_missing_staticfiles(self):
+        # See previous comment about UserSettingsHolder compat code
+        # RemovedInDjango51Warning
+        settings.STORAGES.pop("staticfiles")
+        assert "staticfiles" not in settings.STORAGES
```

Comment:
```suggestion
        assert STATICFILES_STORAGE_ALIAS not in settings.STORAGES
```

Reply:
Is there a reason we're using plain asserts here and above?

```suggestion
        self.assertNotIn(STATICFILES_STORAGE_ALIAS, settings.STORAGES)
```

Reply:
The `settings.STORAGES` can have unexpected values due to custom code coming from `UserSettingsHolder`. Specifically, the test may override the `STORAGES` setting to have a given value, but when executing the tests, it has another value in it. The plain asserts are a way to say "this is assumed to be true" (ie precondition, should never fail) vs using `assertEqual` which states "the test is exercising this condition" (could end up in Failure, particularly when TDDing). Does that make sense?

Reply:
Ah handy to know I didn't realise we did this 👍